refactored.rb

```diff
@@ -0,0 +1,97 @@
+require './school-library/student'
+require './school-library/rental'
+require './school-library/persons'
+require './school-library/teacher'
+require './school-library/book'
+
+class Methods
+  def initialize
+    @person_array = []
+  end
+
+  def create_person
+    puts 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    person_option = Integer(gets.chomp)
+
+    case person_option
+    when 1
+      print 'Age: '
+      user_age = Integer(gets.chomp)
+
+      print 'Name: '
+      user_name = gets.chomp
+
+      print 'Has parent permission? [Y/N]: '
+      user_permission = gets.chomp.to_s.upcase
+
+      case user_permission
+      when 'Y'
+        user_permission = true
+      when 'N'
+        user_permission = false
+      end
+      student = Student.new(user_age, user_name, parent_permission: user_permission)
+      @person_array.push({
+                           output: "[Student] Name: #{student.name}, ID: #{student.id}, Age: #{student.age}",
+                           object: student
+                         })
+
+      puts 'Person created successfully!'
+      puts "\n"
+    when 2
+      print 'Age: '
+      user_age = Integer(gets.chomp)
+
+      print 'Name: '
+      user_name = gets.chomp
```

Comment:
- [ ] In **lines 18 - 22** and **lines 42 to 46** you are requesting Age and Name twice. This defeats the DRY principle. Kindly extract this functionality into a new method and then call this method where necessary.